components/tag-list/tag-list-item-mixin.js

```diff
@@ -140,17 +142,19 @@ export const TagListItemMixin = superclass => class extends LocalizeCoreElement(
 		this.addEventListener('focus', async(e) => {
 			// ignore focus events coming from inside the tag content
 			if (e.composedPath()[0] !== this) return;
+			if (this.keyboardTooltipItem && this.keyboardTooltipShown) {
```

Comment:
This is kinda of leaky in the sense that the `relatedTarget` means something in the context of the `focus` event handler, but it doesn't really mean anything in the context of the tag-list's `d2l-tag-list-item-tooltip-show` event handler. It would be better if we specified a property on `detail` that was more meaningful, but I wonder if we can just handle the announcement here with something like:

```javascript
const tagList = findComposedAncestor(this, elem => elem.tagName === 'D2L-TAG-LIST');
if (!tagList.contains(e.relatedTarget)) {
    // focus moved from outside the tag-list so do the announcement
    announce(...);
}
```
However, I think that `contains(...)` will not identify elements across shadow boundaries, so something like this is probably more resilient.
```javascript
if(!isComposedAncestor(tagList, e.relatedTarget)) {
    // focus moved from outside the tag-list so do the announcement
    announce(...);
}
```
See: [findComposedAncestor](https://github.com/BrightspaceUI/core/blob/main/helpers/dom.js#L57), [isComposedAncestor](https://github.com/BrightspaceUI/core/blob/main/helpers/dom.js#L201).

Then we don't need this extra event dispatch, and the tag-list code is a bit simpler. Thoughts?

Reply:
This is much cleaner, I'll commit the changes.